pkg/jx/cmd/step_create_task.go

```diff
@@ -1431,13 +1431,53 @@ func (o *StepCreateTaskOptions) viewSteps(tasks ...*pipelineapi.Task) error {
 	return nil
 }
 
+func getVersionFromFile(dir string) (string, error) {
+	var version string
+	versionFile := filepath.Join(dir, "VERSION")
+	exist, err := util.FileExists(versionFile)
+	if err != nil {
+		return "", err
+	}
+	if exist {
+		data, err := ioutil.ReadFile(versionFile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to read file %s", versionFile)
+		}
+		text := strings.TrimSpace(string(data))
+		if text == "" {
+			log.Warnf("versions file %s is empty!\n", versionFile)
+		} else {
+			version = text
+			if version != "" {
+				return version, nil
+			}
+		}
+	}
+	return "", errors.New( "failed to read file "+ versionFile)
+}
+
 func (o *StepCreateTaskOptions) setVersionOnReleasePipelines(pipelineConfig *jenkinsfile.PipelineConfig) error {
 	if o.NoReleasePrepare || o.ViewSteps {
 		return nil
 	}
 	version := ""
 
-	if o.PipelineKind == jenkinsfile.PipelineKindRelease {
+	if o.NoApply {
+		version, err := getVersionFromFile(o.Dir)
```

Comment:
Where would the version file come from in this case? Is the idea that one can manually create the file and then it is picked up when your run `step create task` with the `--no-apply` option?

Reply:
What is the intended semantics of  `--no-apply`. From the docs, I would assume that it does all the "normal" things, but then just does not apply the generated resources. But it seems we are not also behaving differently while creating the CRDs?

Reply:
The problem to be fixed is when using *noApply* a version bump happens. That happens in this function _setVersionOnReleasePipelines_. 
In order to fix it, a new option has been added (if noApply) that will not bump the version. It will read the VERSION file that will be created by jx in the initial release. If there is no VERSION file, a default one will be added alongside a warning in the logs. 
The version bump is generated by:

`
Steps: []*syntax.Step{
					{
						Command: "jx step next-version --use-git-tag-only --tag",
						Name:    "next-version",
						Comment: "tags git with the next version",
					},
				},
`
 Having said that, the behaviour is exactly the same for creating the CRDs and the only difference is this function

Reply:
> The problem to be fixed is when using noApply a version bump happens. That happens in this function setVersionOnReleasePipelines.

Why is that an issue? Should the version bump not only depend on the pipeline kind?

> In order to fix it, a new option has been added (if noApply) that will not bump the version.

I am still trying to understand where this becomes an issue. Is this a testing issue or is this an issue in an actual production workflow?

> It will read the VERSION file that will be created by jx in the initial release.

But we never commit the VERSION file, right?

Reply:
the problem is when you run* jx step create task --no-apply* because the version in bumped, and that shouldn't happen. No Apply should change anything.

Reply:
I am wondering whether we are not trying to do drive to distinct things with on option. `--no-apply` should just be about whether the resources are applied to the cluster or not. No circumventing any logic. Then there could be for example a `--dry-run` with the semantics of no modifications to the source/repo. With this option enabled certain side-steps are taken to avoid modifications.

Reply:
I guess this is noapply is trying to achieve, a dry run.